web3.js/src/message/versioned.ts

```diff
@@ -0,0 +1,27 @@
+import {VERSION_PREFIX_MASK} from '../transaction/constants';
+import {Message} from './legacy';
+import {MessageV0} from './v0';
```

Comment:
It's really stressing me out that we're going to have to hard-require the implementation for every version from here to the end of time, but I don't have anything helpful to suggest at this point :)

I wonder if, at some point, we let people create builds (hard forks? 😅) of `web3.js` that no longer include old transaction implementations, if their use case is to _produce_ new transactions and not to _interpret_ old ones.

That said, the existence of this module definitely _encourages_ hard-requiring the whole universe. What would you think about getting rid of this, and instead offering a lightweight `getSerializedMessageVersion()` utility? That way the _end user_ can decide if they want to:

* Hard require every version and switch between them, or
* Fatal/skip if they encounter a version they are not interested in supporting.

Reply:
I'm totally in favor of adding a utility to get the serialized message version but I think it's important to provide the `VersionedMessage.deserialize` utility in addition since anyone deserializing an arbitrary message would need this logic anyways. I think it only makes sense to forgo this utility if you don't need to deserialize messages at all and are only concerned with creating/serializing a particular set of transaction versions. I'm trying to modularize the code that's specific to a particular transaction version so I think we're in decent position for that already.

Reply:
Here's a PR to add that utility function you suggested: https://github.com/solana-labs/solana/pull/27415